### PR TITLE
Update "transform" with "megaevo" battle-engine.js

### PR DESCRIPTION
**Pull request by asdfghjklohhnhn**
This should fix the Mega Evolution and Transform Issue

### CODE REVIEW

File: battle-engine.js [hunk 1]
```diff
@@ -601,6 +601,22 @@ BattlePokemon = (() => {
 	};
 	BattlePokemon.prototype.transformInto = function (pokemon, user, effect) {
 		let template = pokemon.template;
+	
+		if ((user && user.transformed && this.battle.gen >= 5)&&(BattleSide.prototype.chooseMove(willMega))) {
+			this.transformed = false;
+			this.types = this.types
```
Comment (panpawn) [inline battle-engine.js]: not sure i understand the logic here

you also forgot a semi colon on this line

File: battle-engine.js [hunk 2]
```diff
@@ -601,6 +601,22 @@ BattlePokemon = (() => {
 	};
 	BattlePokemon.prototype.transformInto = function (pokemon, user, effect) {
 		let template = pokemon.template;
+
```
Comment (panpawn) [inline battle-engine.js]: there's trailing whitespace on this line

File: battle-engine.js [hunk 3]
```diff
@@ -601,6 +601,22 @@ BattlePokemon = (() => {
 	};
 	BattlePokemon.prototype.transformInto = function (pokemon, user, effect) {
 		let template = pokemon.template;
+	
+		if ((user && user.transformed && this.battle.gen >= 5)&&(BattleSide.prototype.chooseMove(willMega))) {
```
Comment (panpawn) [inline battle-engine.js]: put a space before and after ``&&``

File: battle-engine.js [hunk 4]
```diff
@@ -601,6 +601,22 @@ BattlePokemon = (() => {
...
+			this.knownType = this.side === this.side && this.knownType;
+			for (let statName in this.stats) {
+				this.stats[statName] = this.stats[statName];
+			}
+			this.moveset = [];
+			this.moves = [];
+			this.set.ivs = (this.battle.gen >= 5 ? this.set.ivs : this.set.ivs);
+			this.hpType = (this.battle.gen >= 5 ? this.hpType : this.hpType);
+			this.hpPower = (this.battle.gen >= 5 ? this.hpPower : this.hpPower);
+
```
Comment (panpawn) [inline battle-engine.js]: there's trailing whitespace on this line too

**Comment by panpawn** [comment]
Also, please remember to run ``npm test`` sending in the pull request - this is just to ensure that your changes meet our styling standards.

File: battle-engine.js [hunk 5]
```diff
@@ -601,7 +601,7 @@ BattlePokemon = (() => {
 	};
 	BattlePokemon.prototype.transformInto = function (pokemon, user, effect) {
 		let template = pokemon.template;
-		if ((user && user.transformed && this.battle.gen >= 5) && (BattleSide.prototype.chooseMove(willMega))) {
+		if ((user && user.transformed && this.battle.gen >= 5) && (BattleSide.prototype.chooseMove)) {
```
Comment (panpawn) [inline battle-engine.js]: you can remove the peren around ``BattleSide.prototype.chooseMove`` here :p

File: battle-engine.js [hunk 6]
```diff
@@ -601,6 +601,20 @@ BattlePokemon = (() => {
 	};
 	BattlePokemon.prototype.transformInto = function (pokemon, user, effect) {
 		let template = pokemon.template;
+		if ((user && user.transformed && this.battle.gen >= 5) && (BattleSide.prototype.chooseMove)) {
```
Comment (Zarel) [inline battle-engine.js]: `(BattleSide.prototype.chooseMove)` seems trivially truthy and can be optimized out.

**Comment by Marty-D** [comment]
I'm not entirely sure what this PR is for, since the issue it's trying to fix is already fixed. Apparently someone fixed it and didn't check it off the list or link the commit in the issue.

**Comment by asdfghjklohhnhn** [comment]
@Marty-D somebody fixed the other Hackmons related issue not this one is what Zarel told me

**Comment by asdfghjklohhnhn** [comment]
And yes I just did a test on the server and it does still have this problem

**Pull request closed without merging**